python/tritonserver/_c/tritonserver_pybind.cc

```diff
@@ -2107,6 +2112,7 @@ PYBIND11_MODULE(triton_bindings, m)
       .export_values();
   py::class_<PyServer>(m, "TRITONSERVER_Server")
       .def(py::init<PyServerOptions&>())
+      .def("_ptr", &PyServer::_ptr)
```

Comment:
```suggestion
      .def("_ptr", &PyServer::Ptr)
```

Reply:
This change breaks the build, because now `_ptr` returns an opaque `TRITONSERVER_Server*` through the `PyWrapper<>` instead of the `uintptr_t` it was returning before.

```
...
server/build/_deps/pybind11-src/include/pybind11/detail/descr.h:40:19: error: invalid use of incomplete type 'struct TRITONSERVER_Server'
```

Reply:
is there a way to cast within the pybind definition itself?

Reply:
Does above work or not? This PR has this suggestion applied..

Reply:
If not, just add a lambda wrapper over it.

Reply:
It does not work, @rmccorm4 thank you for assisting in debugging. And thank you for suggestion @GuanLuo , will go ahead and add a lambda wrapper.

Reply:
Actually there are two approaches here:
1) If we write a lambda function, then, the binded function would be static require a parameter and would have to look like:
```.def("_ptr", &[](PyServer* obj){ reinterpret_cast<uintptr_t>(obj->Ptr();)})```. 
In `core/python/_api/_server.py`, we would have to do `Server._ptr(server)`.

2) Instead of this, would it make more sense to keep the original flow of 
```cpp
class PyServer : ... {
     ...
  uintptr_t _ptr() {  
    return reinterpret_cast<uintptr_t>(this->Ptr());
  } 
     ...
}
...
.def("_ptr", &PyServer::_ptr)
```
In `core/python/_api/_server.py`, we would have to do `self.server._ptr()`.

Both options will keep the same external-flow, but was wondering if you had a preference either way for 1 or 2? @GuanLuo

Reply:
Option 1 please, and you don't have to call `Server._ptr(server)` https://github.com/pybind/pybind11/issues/1763#issuecomment-752683926